FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -49,7 +50,28 @@ public enum ModelDownloadType {
 }
 
 /// Downloader to manage custom model downloads.
-public struct ModelDownloader {
+public class ModelDownloader {
+  /// FirebaseApp associated with this instance of ModelDownloader.
+  private let app: FirebaseApp
+
+  private init(app: FirebaseApp) {
+    self.app = app
+  }
+
+  /// Model downloader with default app.
+  static func modelDownloader() throws -> ModelDownloader {
```

Comment:
It pains me to say this, but at the moment we should keep things consistent with the other Firebase SDKs and `fatalError` instead of throwing.

We'll need to re-evaluate, but we treat this in all other SDKs as a programmer error that isn't recoverable.

Reply:
Also this should be `public`.

Reply:
Good catch, updated.